kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -77,7 +77,7 @@ class KyuubiSessionImpl(
   lazy val engine: EngineRef = new EngineRef(
     sessionConf,
     user,
-    sessionManager.groupProvider.primaryGroup(user, optimizedConf.asJava),
```

Comment:
this is also a bugfix as using optimized conf is wrong here

Reply:
Yeah, actually it seems that conf is not used in subsequent calls ? 
```
# HadoopGroupProvider.scala

  override def groups(user: String, sessionConf: JMap[String, String]): Array[String] =
    UserGroupInformation.createRemoteUser(user).getGroupNames match {
      case Array() =>
        warn(s"There is no group for $user, use the client user name as group directly")
        Array(user)
      case groups => groups
    }

```